homeassistant/components/moehlenhoff_alpha2/climate.py

```diff
@@ -0,0 +1,152 @@
+"""Support for Alpha2 room control unit via Alpha2 base."""
+import logging
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    HVAC_MODE_AUTO,
+    PRESET_COMFORT,
+    PRESET_ECO,
+    PRESET_NONE,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from . import DOMAIN, SIGNAL_HEATAREA_DATA_UPDATED
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add Alpha2Climate entities from a config_entry."""
+
+    base_update_handler = hass.data[DOMAIN][config_entry.entry_id]["connection"]
+    await base_update_handler.async_update()
+
+    entities = []
+    for heatarea in base_update_handler.base.heatareas:
+        entities.append(Alpha2Climate(base_update_handler, heatarea))
+    async_add_entities(entities)
+
+
+# https://developers.home-assistant.io/docs/core/entity/climate/
+class Alpha2Climate(ClimateEntity):
+    """Alpha2 ClimateEntity."""
+
+    target_temperature_step = 0.2
+
+    _attr_should_poll = False
+    _attr_supported_features = SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+    _attr_hvac_mode = HVAC_MODE_AUTO
+    _attr_hvac_modes = [HVAC_MODE_AUTO]
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_preset_modes = [PRESET_NONE, PRESET_COMFORT, PRESET_ECO]
+
+    def __init__(self, base_update_handler, data):
+        """Initialize Alpha2 ClimateEntity."""
+        self._base_update_handler = base_update_handler
+        self._data = data
+
+    async def async_added_to_hass(self):
+        """Register callbacks."""
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                SIGNAL_HEATAREA_DATA_UPDATED,
+                self._handle_heatarea_data_updated,
+            )
+        )
+
+    def _handle_heatarea_data_updated(self, data):
+        """Handle updated heatarea data."""
+        if data["NR"] == self._data["NR"]:
+            self._data = data
+            self.async_schedule_update_ha_state()
+
+    @property
+    def name(self):
+        """Return the name of the climate device."""
+        return self._data["HEATAREA_NAME"]
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return float(self._data.get("T_TARGET_MIN", 0))
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return float(self._data.get("T_TARGET_MAX", 30))
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return float(self._data.get("T_ACTUAL", 0))
+
+    def set_hvac_mode(self, hvac_mode: str):
+        """Set new target hvac mode."""
+        pass
```

Comment:
We usually leave it empty.

Examples: 

https://github.com/home-assistant/core/blob/24c1256c2c8bef2d39a2e483ac09689d850181a0/homeassistant/components/velbus/climate.py#L83-L84

https://github.com/home-assistant/core/blob/24c1256c2c8bef2d39a2e483ac09689d850181a0/homeassistant/components/xs1/climate.py#L112-L114
https://github.com/home-assistant/core/blob/24c1256c2c8bef2d39a2e483ac09689d850181a0/homeassistant/components/toon/climate.py#L122-L125
```suggestion
```